maintenance/wikia/generateMessageFile.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+if ( !defined( 'MEDIAWIKI' ) ) {
+	ini_set( "include_path", dirname(__FILE__)."/.." );
+	require_once( 'commandLine.inc' );
+}
+
+$USAGE = 
+	"Usage: php generateLangFile.php [--lang=(all|ja|zh|zh-tw|pl|...)] [--removedb=(no|yes)] [--merge=(no|yes)] [--filename=Messages%s.php] [--dir=/tmp] | --help\n" .
+	"\toptions:\n" .
+	"\t\t--help	show this message\n" .
+	"\t\t--lang	language code (default: 'all': generate all messages)\n" .
+	"\t\t--removedb	remove articles from database, default: no\n".
+	"\t\t--merge	merge with existing wikia messages, default: no\n".
+	"\t\t--filename	format of output filename, default: Messages%s.php (where '%s' is a --lang parameter)\n".
+	"\t\t--dir	 output dir name, default: '/tmp' \n";
+
+if ( in_array( '--help', $argv ) ) {
+	wfDie( $USAGE );
+}
+
+$params = @$options;
+$lang = (array_key_exists('lang', $params) && !empty($params['lang'])) ? $params['lang'] : 'all';
+$removedb = (array_key_exists('removedb', $params) && !empty($params['removedb'])) ? $params['removedb'] : 'no';
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "removedb" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)